### ISSUE TITLE
Check for single field regions

### ISSUE THREAD

**Issue by tidyui** [PiranhaCMS/piranha.core.analyzers#9]
Forgive me if this is something you've already implemented, but one thing that would be great to check for is wrong implementation of single field regions (this is because if you implement it wrong the transformation will fail).

## Correct implementation

~~~ csharp
public class MyPage : Page<MyPage>
{
    [Region]
    public HtmlField Hero { get; set; }
}
~~~

## Incorrect implementation

~~~ csharp
public class HeroRegion
{
    [Field]
    public HtmlField Body { get; set; }
}

public class MyPage : Page<MyPage>
{
    [Region]
    public HeroRegion Hero { get; set; }
}
~~~

So in other words, regions made up of objects with a **single field** is **not** valid.

**Comment by mikaellindemann** [comment]
Would you consider this a warning or an error?

I guess, if it just doesn't work then it could be implemented as giving a compiler error to be fixed.

**Comment by tidyui** [comment]
This is considered an error, and it’s also clearly stated in the docs. But some extra compiler help could be useful 😁

**Comment by mikaellindemann** [comment]
And how should it handle the field types that, according to documentation, does not support/handle single field regions well? (E.g CheckBoxField)
